app/backend/core/authentication.py

```diff
@@ -0,0 +1,164 @@
+# Refactored from https://github.com/Azure-Samples/ms-identity-python-on-behalf-of
```

Comment:
High-level observation: there's a lot going on in this code. Does any of this belong in a different SDK? And did you check if https://github.com/rayluo/identity does any of this for you? It's an msal abstraction for login.

Reply:
this also has the "DISCLAIMER: The code in this repo is not officially supported by Microsoft and is not intended for production use. " in the description. I'm happy to look at it, but i used an existing sample for this code

Reply:
Oh, interesting, we changed the learn.com tutorials to use that repo. I'll ask Ray the status of that.

Reply:
i took a brief look at this and did not see the functionality that i needed (claims parsing, on behalf of flow), but we can look again if required.